### DISCUSSION TITLE
Wsl2g distribution

### DISCUSSION THREAD

**Discussion by evilz** [ravynsoft/ravynos#232] [Ideas]
Hello all,

Do you think it's possible to create a version to get it run on windows subsystem for Linux with GUI???

It could be awesome 😎

**Comment by markshank** [comment]
I think it would require a windows subsystem for FreeBSD first.

**Comment by davidchisnall** [comment]
WSL2 is mostly Hyper-V and all of the guest parts are open source.  The GUI support is done by running a Linux VM that provides a Wayland compositor and using PipeWire from other WSL2 VMs to send audio and video to it.  I believe FreeBSD supports PipeWire and so it may be possible to use this directly.  The file sharing is done via 9p-over-vmbus.  I believe FreeBSD now has a 9p-over-VirtIO driver (and a VMBus driver), so adding support for this is just a matter of plumbing (probably a few days of kernel hacking).

There's also a bunch of integration work, such as an image activator (Linux calls this a binfmt driver) that takes PE/COFF files and passes their paths via a Hyper-V socket to the host so that you can run .exe files.

**Reply by mszoek** [reply]
Very interesting- thanks David! Maybe this is more feasible than I thought. I doubt I can take on any more at the moment (and I don't have Windows) but perhaps someone else could if there's demand here.
